docs/designs/xref.md

```diff
@@ -4,6 +4,152 @@ Besides using file path to link to another file, DocFX also allows you to give a
 - API reference documentation is usually auto generated so it's difficult to find its file path.
 - You want to reference to files in another project without need to know its file structure.
 
+## Feature Requirements/Scenarios
+- Define an internal UID in `.md` and reference to this UID within the current repository
+  ```yaml
+  inputs:
+    docs/a.md: |
+    ---
+    title: Title from yaml header a
+    uid: a
+    description: some description
+    ---
+    docs/b.md: Link to @a
+  outputs:
+    docs/b.json: |
+        {"conceptual":"<p>Link to <a href=\"a\">Title from yaml header a</a></p>\n"}
+  ```
+  > **_Notice_**: Only title will be considered as xref properrty for `uid` definition in `.md` files
+- Define multiple UID with same value internally with different versions and reference to this UID without versioning within the current repository
+    - If all versions for this UID are within the same product, take the one with highest versioning respecting the referencing file
+      ```yaml
+        # v1: 1.0, 2.0, 3.0
+        # v2: 4.0, 5.0
+        # v3: 3.0 
```

Comment:
> v3: 3.0  [](start = 9, length = 9)

what if the monikers of v3 folder is `4.0`

Reply:
we can come up with more tests when the PR is out